strax/processing/hitlets.py

```diff
@@ -0,0 +1,372 @@
+    # we extended our hitlet into regions without any data so we have to chop
+    # "time" according to the data we found....
+    time = hitlet['time'] + data_start * hitlet['dt']
+    temp_data = temp_data[data_start:data_end] + r['baseline'] % 1
+    return temp_data, time
+
+
+@numba.njit(nogil=True, cache=True)
+def _get_thing_data(thing, container):
+    """
+    Function which returns data for some overlapping indices of a thing
+    in a container. 
+    
+    Note:
+        Thing must be of the interval dtype kind.
+    """
+    overlap_hit_i, overlap_record_i = strax.overlap_indices(thing['time']//thing['dt'],
+                                                            thing['length'],
+                                                            container['time']//container['dt'],
+                                                            container['length'])
+    data = container['data'][overlap_record_i[0]:overlap_record_i[1]]
+    return data, overlap_hit_i
+
+# ----------------------
+# Hitlet splitting:
+# ----------------------
+@export   # <-- Required in peak_splitting but not by the user
+def _update_new_hitlets(hitlets, records, next_ri, to_pe):
+    """
+    Function which computes the hitlet data area and record_i after 
+    splitting.
+    """
+    _update_record_i(hitlets, records, next_ri)
+    get_hitlets_data(hitlets, records, to_pe)
+
+
+@numba.njit(cache=True, nogil=True)
+def _update_record_i(new_hitlets, records, next_ri):
+    """
+    Function which updates the record_i value of the new hitlets. 
+    
+    Notes:
+        Assumes new_hitlets to be sorted in time.
+    """
+    for ind, hit in enumerate(new_hitlets):
+
+        updated = False
+        counter = 0
+        last_ri = 0
+        current_ri = hit['record_i']
+        while not updated:
+            r = records[current_ri]
+            # Hitlet must only partially be contained in record_i:
+            time = hit['time']
+            end_time = strax.endtime(hit)
+            start_in = (r['time'] <= time) & (time < strax.endtime(r))
+            end_in = (r['time'] < end_time) & (end_time <= strax.endtime(r))
+            if start_in or end_in:
+                hit['record_i'] = current_ri
+                updated = True
+                break
+            else:
+                last_ri = current_ri
+                current_ri = next_ri[current_ri]
+                counter += 1
+                
+            if current_ri == -1:
+                print('Record:\n', r, '\nHit:\n', hit)
+                raise ValueError('Was not able to find record_i')
+
+            if counter > 100:
+                print(ind, last_ri)
+                raise RuntimeError('Tried too often to find correct record_i.')
+
+
+# ----------------------
+# Hitlet properties:
+# ----------------------
+@export
+@numba.njit(cache=True, nogil=True)
+def get_fwxm(data, index_maximum, percentage=0.5):
```

Comment:
Could you not drop the ```index_maximum``` and do a ```index_maximum=np.argmax(data)``` in the function itself

Reply:
Sorry, the export should not have been there. This function is only called within hitlet_properties

Reply:
sure but it's easy if people (or you) want to use it via strax.processing.hitlets.fet_fwxm. If the arguments can be made simpler it allows for easier maintenance as you can also use the function elsewhere